src/v/kafka/server/tests/quota_manager_test.cc

```diff
@@ -223,3 +223,78 @@ SEASTAR_THREAD_TEST_CASE(static_config_test) {
         BOOST_CHECK_EQUAL(it->second->pm_rate->rate(), 1026);
     }
 }
+
+SEASTAR_THREAD_TEST_CASE(update_test) {
+    using clock = kafka::quota_manager::clock;
+    fixture f;
+    f.start().get();
+    auto stop = ss::defer([&] { f.stop().get(); });
+
+    set_config(basic_config).get();
+
+    auto now = clock::now();
+    {
+        // Update fetch config
+        ss::sstring client_id = "franz-go";
+        f.sqm.local().record_fetch_tp(client_id, 8194, now).get();
+        f.sqm.local()
+          .record_produce_tp_and_throttle(client_id, 8192, now)
+          .get();
+
+        set_config([](config::configuration& conf) {
+            auto fetch_config = YAML::Load(std::string(raw_basic_fetch_config));
+            for (auto n : fetch_config) {
+                n["quota"] = n["quota"].as<uint32_t>() + 1;
+            }
+            conf.kafka_client_group_fetch_byte_rate_quota.set_value(
+              fetch_config);
+        }).get();
```

Comment:
Config update is done async here. I'm wondering if this test will be flaky. Perhaps we should add a delay here to increase the chance of the async update to propagate.

Reply:
> Config update is done async here. I'm wondering if this test will be flaky.

Yeah, sometimes I wish config updates could be async.

> Perhaps we should add a delay here to increase the chance of the async update to propagate.

A fair point. I'll add a yield or something.